File: src/main/java/org/flighthub/Repository/IFlightRepository.java
package org.flighthub.Repository;

import org.flighthub.Domain.Entity;
import org.flighthub.Domain.Flight;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Optional;
import java.util.UUID;

public interface IFlightRepository extends Repository<UUID, Flight> {
    Optional<Flight> findOne(UUID id);

    Iterable<Flight> findByDestinationAndDepartureDate(String destination, LocalDate departureDate);

    Iterable<Flight> findAll();

    Flight save(Flight entity);

    Optional<Flight> delete(UUID id);

    Optional<Flight> update(Flight entity);

     void updateAvailableSeats(UUID flightId, int newAvailableSeats);
}

File: Server - C#/persistence/IFlightRepository.cs
using FlightHubC_.Domain;
using FlightHubC_.Repository;
using System.Collections.Generic;
using System;
namespace FlightHubC_.Repository
{
    public interface IFlightRepository : IRepository<Guid, Flight>
    {
        Flight FindOne(Guid id);
        IEnumerable<Flight> FindByDestinationAndDepartureDate(string destination, DateTime departureDate);

        IEnumerable<Flight> FindAll();
        Flight Save(Flight entity);
        Flight Delete(Guid id);
        Flight Update(Flight entity);
    }
}